Scripts/Runtime/Entities/TaskSystem/AbstractTaskDriver.cs

```diff
@@ -48,35 +54,70 @@ public abstract class AbstractTaskDriver : AbstractAnvilBase
         /// </summary>
         public AbstractTaskSystem TaskSystem { get; }
         
+        /// <summary>
+        /// Reference to the associated <see cref="World"/>
+        /// </summary>
+        public World World { get; }
+
         internal CancelRequestsDataStream CancelRequestsDataStream { get; }
+        internal List<AbstractTaskStream> TaskStreams { get; }
+        internal TaskDriverCancellationPropagator CancellationPropagator { get; private set; }
 
         protected AbstractTaskDriver(World world, AbstractTaskSystem abstractTaskSystem)
         {
-            TaskSystem = abstractTaskSystem;
+            World = world;
```

Comment:
Here is where you wouldn't need to pass in the world. Would avoid potential errors where `world != abstractTaskSystem.World`

Reply:
See: https://github.com/decline-cookies/anvil-unity-dots/pull/87/files#r995078637

We wouldn't ever get `world != abstractTaskSystem.World` because the typed generic version creates it for us:

`protected AbstractTaskDriver(World world) : base(world, world.GetOrCreateSystem<TTaskSystem>())`

Reply:
I see, I mean there's an off chance if someone else sub-classes this.
If you change it to
`protected AbstractTaskDriver(World world) : base(world.GetOrCreateSystem<TTaskSystem>())`
then it's guaranteed.

Reply:
I can't do that unfortunately because I need both `World` and the `TaskSystem` and there's no guarantee that the `TaskSystem` has it's world yet from this chain https://github.com/decline-cookies/anvil-unity-dots/pull/87#discussion_r995078637

Reply:
Addressed: See https://github.com/decline-cookies/anvil-unity-dots/pull/87#discussion_r995893521